services/blog_post_processor.py

```
@@ -28,6 +28,65 @@ def _get_twitch_embed_domains(self) -> str:
+            
+            # Additional validation: ensure it's not just dots or hyphens
+            if domain.replace('.', '').replace('-', '').replace(':', '') == '':
+                raise ValueError(f"Invalid domain: '{domain}' contains only special characters")
+            
+            # Validate port range if present
+            if ':' in domain:
+                try:
+                    port_str = domain.split(':')[-1]
+                    port = int(port_str)
+                    if not (1 <= port <= 65535):
+                        raise ValueError(f"Invalid port range: '{port}'. Port must be between 1 and 65535.")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)